src/Engine/ProtoCore/RuntimeStatus.cs

```diff
@@ -52,13 +52,31 @@ public struct WarningEntry
             public int AstID;
             public string Filename;
         }
+
+        public enum InfoID
+        {
+            Default
+        }
+
+        public struct InfoEntry
+        {
+            public Runtime.InfoID ID;
+            public string Message;
+            public int Line;
+            public int Column;
+            public int ExpressionID;
+            public Guid GraphNodeGuid;
+            public int AstID;
+            public string Filename;
+        }
```

Comment:
I wish we didn’t have to expose more APIs from the engine level

Reply:
I did look at it but unfortunately because the current `public interface ILiveRunner` setup, the function involved needs to be public and also the return object as well. VS does throw errors if trying to make this internal

Reply:
Is this really a good candidate for a struct?

Reply:
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct

Reply:
@mjkkirschner I am not sure but maybe we could break these changes in 3.0. Right now I am just following the previous design for all these engine related code